### ISSUE TITLE
ProviderError: Exceeded maximum write data: max bytes 25000

### ISSUE THREAD

**Issue by knerushkin-rumblefish** [nervosnetwork/godwoken-web3#312]
Contracts deployed:
[Balancer Vault](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/vault/contracts/Vault.sol)

Can't deploy contract because of contract size
`ProviderError: Exceeded maximum write data: max bytes 25000, writen bytes 25327`
![balancer-contract-size-error](https://user-images.githubusercontent.com/82823678/166914400-b65c1c81-6dfc-4773-826b-5260132781cd.png)

No problems on Ropsten:
![balancer-deployment-vault-ropsten](https://user-images.githubusercontent.com/82823678/166914412-608a6a50-e874-42a9-ae0d-daff68432ba9.png)

Deployment is performed with hardhat
_hardhat.config_
```ts
{
  solidity: {
    version: '0.7.1',
    settings: {
      optimizer: { enabled: true, runs: 10 }
    }
  },
  networks: {
    "godwoken-testnet": {
      accounts: [...],
      chainId: 0x315db00000006,
      url: "https://godwoken-testnet-web3-v1-rpc.ckbapp.dev",
      throwOnCallFailures: true,
      gas: 12000000,
      timeout: 100000,
      gasMultiplier: 2
   }
} 
```

**Comment by keroro520** [comment]
As an optimistic rollup layer2, Godwoken stores layer2 transactions in layer1 chain. We know layer1's storage is expensive and limited. Therefore, Godwoken has to limit the written data of layer2 transactions.
In Godwoken, there is a hard-coded 25000bytes size limit for storing data.
That means it is not allowed to:
  - Deploy a contract with a size greater than 25000 bytes
  - `SSTORE` an entry larger than 25000 bytes

**Comment by Flouse** [comment]
## Related issue
- https://github.com/nervosnetwork/godwoken-polyjuice/issues/108#issuecomment-989479759

**Comment by RetricSu** [comment]
can you try with hardhat config: 

```sh
{
  solidity: {
    version: '0.7.1',
    settings: {
      optimizer: { enabled: true, runs: 200 }
    }
  },
  networks: {
    "godwoken-testnet": {
      accounts: [...],
      chainId: 0x315db00000006,
      url: "https://godwoken-testnet-web3-v1-rpc.ckbapp.dev",
      throwOnCallFailures: true,
      gas: 12000000,
      timeout: 100000,
      gasMultiplier: 2
   }
} 
```

to see if contract code size can be reduced below 24k?

@knerushkin-rumblefish

**Comment by knerushkin-rumblefish** [comment]
@RetricSu I've try with `runs: 200` but in this case we get worse optimization than in case on `runs: 10`. So unfortunately I've no idea how to reduce size with solidity optimizer. It's look like contract should be modified, to make it work on godwoken-testenet-v1

**Comment by knerushkin-rumblefish** [comment]
@RetricSu just strange that those contract's works on optimistic rollups like Arbitrum, maybe they have some special deployment process for it

**Comment by knerushkin-rumblefish** [comment]
@RetricSu Now I'm remove part of contract features and get  contract size 23.049, I'm loging it now with hardhat-contract-sizer. And still problem to deploy it on godwoken, working fine for ropsten

**Comment by RetricSu** [comment]
> Now I'm remove part of contract features and get contract size 23.049

can you paste the contract artifact? I find out that the error happened because `eth_estimateGas` request failed before the deployment. I want to use your contract artifact to debug in local env.

**Comment by knerushkin-rumblefish** [comment]
[Vault.tar.gz](https://github.com/nervosnetwork/godwoken-web3/files/8650146/Vault.tar.gz)

**Comment by RetricSu** [comment]
> There id hardhat artifact [Vault.tar.gz](https://github.com/nervosnetwork/godwoken-web3/files/8650146/Vault.tar.gz) Is it enough?

sorry, my mistake. I check out the original balancer repo and I found out that the contract is kind of complex for deployment constructor, so I am wondering if you have any repo for me to reproduce this issue, a repo that can be directly used to deploy this contract on godwoken testnet or devnet.

**Comment by knerushkin-rumblefish** [comment]
Yeap, give me a moment, I'll prepare

**Comment by knerushkin-rumblefish** [comment]
Pls check this out
[Error Reproduced](https://github.com/knerushkin-rumblefish/exceeded-maximum-write-data)

**Comment by knerushkin-rumblefish** [comment]
@RetricSu Just realise that I'm trying to add gasLimit through hardhat, and gasLimit do not set on transaction, so I'm add manual gasLimit to deploy call. And now it's behave differently, it's stuck
I'm add update to [Repo](https://github.com/knerushkin-rumblefish/exceeded-maximum-write-data)

**Comment by RetricSu** [comment]
> @RetricSu I'm just realized that I'm trying to add gasLimit through hardhat, and gasLimit do not set on transaction, so I'm add manual gasLimit to deploy call. And now it's behave differently, it's stuck I'm add update to [Repo](https://github.com/knerushkin-rumblefish/exceeded-maximum-write-data)

yes I can confirm on my local godwoken devnet, I have reproduce this behavior. 

and I noticed that the contract size `Vault  22.625` is smaller than 25000, but it is the deployed-byte-code size aka run-time contract byte code size. the contract with constructor is greater than 25000, and I guess that might trigger some error with exceed writing limit. we have to track the polyjuice to find out.

as for the stuck issue, it should be another issue, in most time, should return a error message. log on my machine:

```sh
godwoken_1                        | 2022-05-09T11:09:01.823289Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: BEGIN store data key    
godwoken_1                        | 2022-05-09T11:09:01.823854Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: code_data_hash 0x57b5ac507ac3a332704e8a0424e147d57556b9b422139d5bf860eaddd336e21c    
godwoken_1                        | 2022-05-09T11:09:01.839139Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: BEGIN store data    
godwoken_1                        | 2022-05-09T11:09:01.839773Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: contract_code_len => 22625    
godwoken_1                        | 2022-05-09T11:09:01.844331Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: END store data    
godwoken_1                        | 2022-05-09T11:09:01.844577Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: [handle_message] used_memory(Bytes) => 22656    
godwoken_1                        | 2022-05-09T11:09:01.844656Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: [handle_message] gas left => 4398337    
godwoken_1                        | 2022-05-09T11:09:01.844716Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: [handle_message] status_code => 0    
godwoken_1                        | 2022-05-09T11:09:01.849125Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: gas limit => 4500000    
godwoken_1                        | 2022-05-09T11:09:01.849300Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: gas left => 4398337    
godwoken_1                        | 2022-05-09T11:09:01.849359Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: gas price => 1    
godwoken_1                        | 2022-05-09T11:09:01.849506Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: fee => 101663    
godwoken_1                        | 2022-05-09T11:09:01.849606Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: [run_polyjuice] used_memory(Bytes) => 22656    
godwoken_1                        | 2022-05-09T11:09:01.851164Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract syscall: SYS_PAY_FEE] payer: 966b30e576a4d6731996748b48dd67c94ef29067, registry_id: 2, sudt_id: 1, amount: 101663    
godwoken_1                        | 2022-05-09T11:09:01.851338Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract syscall: SYS_PAY_FEE] payer: 966b30e576a4d6731996748b48dd67c94ef29067, registry_id: 2, sudt_id: 1, amount: 101663    
godwoken_1                        | 2022-05-09T11:09:01.851388Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::syscalls: [contract debug]: [run_polyjuice] finalize    
godwoken_1                        | 2022-05-09T11:09:01.852993Z DEBUG push_transaction:push_transaction_with_db{tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:execute_tx:unchecked_execute_transaction{block=645 tx_hash=Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a)}:machine_run{backend=Polyjuice}: gw_generator::generator: [execute tx] VM machine_run time: 111ms, exit code: 0 used_cycles: 5256573    
godwoken_1                        | 2022-05-09T11:09:01.855069Z  INFO gw_rpc_server::registry: push Tx Byte32(0xa4d74d11b9976b38f914565bd8dd09652f3004be58279527b7e807ff77bf810a) failed Exceeded maximum write data: max bytes 25000, writen bytes 25187    
```

**Comment by knerushkin-rumblefish** [comment]
@RetricSu So did this problem can be resolved, and did it going to be fixed in next godwoken release, or maybe it can be hotfixed?

**Comment by RetricSu** [comment]
> @RetricSu So did this problem can be resolved, and did it going to be fixed in next godwoken release, or maybe it can be hotfixed?

we need to track more detail in polyjuice. will update if I got any more progress.

**Comment by knerushkin-rumblefish** [comment]
@RetricSu Thanks
btw after deploy stuck I get 
```bash
HeadersTimeoutError: Headers Timeout Error
    at Timeout.onParserTimeout [as _onTimeout] (/home/all-mighty-func/dev/hadouken-project/stableswap-v2/node_modules/undici/lib/client.js:1005:28)
    at listOnTimeout (internal/timers.js:559:11)
    at processTimers (internal/timers.js:500:7) {
  code: 'UND_ERR_HEADERS_TIMEOUT',
  transactionHash: '0xd67350b08469519eb3848b27db2bb65e5bd9793bbb8f43d14983ad0ffb51b39e'
}
Done in 4910.08s.

```

**Comment by RetricSu** [comment]
the contract will create one sub contract which has 2562 bytes (22625 + 2562 = 25187), two contracts trigger an error on godwoken, fix pr:

- https://github.com/nervosnetwork/godwoken/pull/685

**Comment by knerushkin-rumblefish** [comment]
@RetricSu So can I test deployment localy? maybe I can set up godwoken local with this PR? Or it can be checked straight on testnet?

**Comment by RetricSu** [comment]
> @RetricSu So can I test deployment localy? maybe I can set up godwoken local with this PR? Or it can be checked straight on testnet?

you can test it on local devnet, testnet might need some time to deploy

**Comment by Flouse** [comment]
> So can I test deployment localy? maybe I can set up godwoken local with this PR?

I think this document may help.
https://github.com/RetricSu/godwoken-kicker/blob/compatibility-changes/docs/manual-build.md

**Comment by knerushkin-rumblefish** [comment]
Unfortunatly deployment process is generally failing but from time to time it's succeed for those "big" contracts, but on those deployed contracts on local godwoken kicker 
```yaml
# Requires docker-compose >= 1.29.0
version: '3.8'

services:
  postgres:
    image: postgres:alpine
    ports: 
      - 5432:5432
    environment:
      POSTGRES_USER: user
      POSTGRES_DB: lumos
      POSTGRES_PASSWORD: password
    volumes:
      - ./postgres/web3-indexer-schema.sql:/docker-entrypoint-initdb.d/web3-indexer-schema.sql
      - ./postgres/data:/var/lib/postgresql/data
    
  redis:
    image: redis:bullseye
    user: root:root
    ports: 
      - 6379:6379
    volumes:
      - ./redis/data:/data

  ckb:
    image: nervos/ckb:v0.103.0
    user: root
    ports:
      - 8114:8114   # rpc
      - 8115:8115   # p2p network
    volumes:
      - ./layer1/ckb:/var/lib/ckb
    command: [ "run", "-C", "/var/lib/ckb" ]

  ckb-miner:
    image: nervos/ckb:v0.103.0
    user: root
    volumes:
      - ./layer1/ckb:/var/lib/ckb
    command: [ "miner", "-C", "/var/lib/ckb" ]

  # "nervos/ckb" image have no http clients to do health check for CKB.
  # This short-term service act as a workaround to do health check.
  check-ckb-started-successfully:
    image: curlimages/curl
    volumes:
      - ./layer1/ckb:/var/lib/ckb
    command: [ 'http://ckb:8114', '-H', 'content-type: application/json', '-d', '{ "id": 2, "jsonrpc": "2.0", "method": "get_tip_block_number", "params": [] }' ]
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 1000
    depends_on:
      - ckb

  ckb-indexer:
    image: nervos/ckb-indexer:0.3.2
    environment:
      CKB_RPC: http://ckb:8114
    ports:
      - "8116:8116"
    volumes:
      - ./ckb-indexer:/var/lib/ckb-indexer
    command: [ "-c", "http://ckb:8114", "-l", "0.0.0.0:8116", "-s", "/var/lib/ckb-indexer/data" ]
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 10
    depends_on:
      check-ckb-started-successfully:
        condition: service_completed_successfully

  godwoken:
    image: ghcr.io/flouse/godwoken-prebuilds:dev
    healthcheck:
      test: /var/lib/layer2/healthcheck.sh
      start_period: 10s
      interval: 30s
      retries: 600
    environment:
      RUST_LOG: info,gw_generator=debug
      GODWOKEN_MODE: fullnode
      RUST_BACKTRACE: full
      STORE_PATH: /var/lib/layer2/data
      ACCOUNTS_DIR: /accounts
      POSTGRES_USER: user
      POSTGRES_DB: lumos
      POSTGRES_PASSWORD: password
      GITHUB_RUN_ID: ${GITHUB_RUN_ID:-""}
    volumes:
      - ./layer2:/var/lib/layer2
      - ../accounts:/accounts
    ports:
      - 8119:8119 # rpc
      - 8120:8120 # err_receipt_ws_listen
    entrypoint: "/var/lib/layer2/entrypoint.sh"
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 1
    depends_on:
      postgres:
        condition: service_started
      ckb:
        condition: service_started
      ckb-miner:
        condition: service_started
      ckb-indexer:
        condition: service_started
      check-ckb-started-successfully:
        condition: service_completed_successfully

  web3:
    image: ghcr.io/nervosnetwork/godwoken-web3-prebuilds:compatibility-breaking-changes-2b1dc8e
    healthcheck:
      test: curl http://127.0.0.1:8024 || exit 1
      start_period: 10s
      interval: 10s
      retries: 30
    volumes: 
      - ./layer2/config:/var/lib/layer2/config
      - ./web3/entrypoint.sh:/var/lib/web3/entrypoint.sh
    environment: 
      CONFIG_DIR: /var/lib/layer2/config
    entrypoint: /var/lib/web3/entrypoint.sh
    ports:
      - 8024:8024
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 10
    depends_on: 
      ckb:
        condition: service_started
      ckb-miner:
        condition: service_started
      check-ckb-started-successfully:
        condition: service_completed_successfully
      godwoken:
        condition: service_healthy
      postgres:
        condition: service_started
      redis:
        condition: service_started

  web3-indexer:
    image: ghcr.io/nervosnetwork/godwoken-web3-indexer-prebuilds:compatibility-breaking-changes-b97b1a5
    volumes:
      - ./layer2/config:/var/lib/layer2/config
      - ./web3/indexer_entrypoint.sh:/var/lib/web3-indexer/entrypoint.sh
    working_dir: /var/lib/web3-indexer
    entrypoint: /var/lib/web3-indexer/entrypoint.sh
    environment:
      CONFIG_DIR: /var/lib/layer2/config
      RUST_LOG: info
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 10
    depends_on: 
      web3:
        condition: service_started
      godwoken:
        condition: service_healthy
      postgres:
        condition: service_started
      redis:
        condition: service_started


```
now I do not able to perform wait on transaction object.
```bash 
(node:352324) UnhandledPromiseRejectionWarning: Error: transaction failed [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (transactionHash="0x497db208e8cc885ae2f889f8cf1404b6b330bed616c6c9267e4d46a446aea53f", transaction={"hash":"0x497db208e8cc885ae2f889f8cf1404b6b330bed616c6c9267e4d46a446aea53f","type":0,"accessList":null,"blockHash":"0xd902b02bc93175ecef7248239c58f8f8849436de4f879628b43c4277bb417654","blockNumber":5443,"transactionIndex":0,"confirmations":5,"from":"0x0000000000000000000000000000000000000000","gasPrice":{"type":"BigNumber","hex":"0x01"},"gasLimit":{"type":"BigNumber","hex":"0x1c4f"},"to":null,"value":{"type":"BigNumber","hex":"0x00"},"nonce":0,"data":"0x","r":"0x0000000000000000000000000000000000000000000000000000000000000000","s":"0x0000000000000000000000000000000000000000000000000000000000000000","v":0,"creates":"0xBd770416a3345F91E4B34576cb804a576fa48EB1","chainId":0}, receipt={"to":null,"from":"0x0000000000000000000000000000000000000000","contractAddress":null,"transactionIndex":0,"gasUsed":{"type":"BigNumber","hex":"0x1c4f"},"logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","blockHash":"0xd902b02bc93175ecef7248239c58f8f8849436de4f879628b43c4277bb417654","transactionHash":"0x497db208e8cc885ae2f889f8cf1404b6b330bed616c6c9267e4d46a446aea53f","logs":[],"blockNumber":5443,"confirmations":5,"cumulativeGasUsed":{"type":"BigNumber","hex":"0x1c4f"},"status":0,"type":0,"byzantium":true}, code=CALL_EXCEPTION, version=providers/5.6.6)
```

 Also in logs web3 I'm getting 
```bash
web3_1                            | [1] 2022-05-13T08:39:46.710Z [error]: Redis Client Error Error: getaddrinfo EAI_AGAIN redis
web3_1                            | [1]     at GetAddrInfoReqWrap.onlookup [as oncomplete] (dns.js:71:26) {
web3_1                            | [1]   errno: -3001,
web3_1                            | [1]   code: 'EAI_AGAIN',
web3_1                            | [1]   syscall: 'getaddrinfo',
web3_1                            | [1]   hostname: 'redis'
web3_1                            | [1] }
```

Maybe my local kicker configuration is not correct?

**Comment by knerushkin-rumblefish** [comment]
With wait failure it's my fault I've debugged that I'm using wrong roles. But problem with deployment still occur that it's rather failing and sometimes succeed

**Comment by knerushkin-rumblefish** [comment]
Btw, looks like error's from contract are not visible on client side

**Comment by RetricSu** [comment]
testnet have been updated, you should be able to deploy contract with 24k size limit. is there any trouble? @knerushkin-rumblefish

**Comment by knerushkin-rumblefish** [comment]
@RetricSu Yeap on current v1.1 no problem with deployment
